types/src/transaction/transaction.rs

```diff
@@ -163,6 +165,52 @@ impl Transaction {
             } => vec![*shard_id],
         }
     }
+
+    pub fn verify(&self) -> Result<(), Error> {
+        match self {
+            Transaction::AssetTransfer {
+                inputs,
+                outputs,
+                ..
+            } => {
+                // FIXME: check burns
```

Comment:
@majecty I'm just curious. Which verifications should we do here?

Reply:
Check whether inputs and burns have common input or not.